CaloMC/src/CaloDigiMaker_module.cc

```diff
@@ -140,31 +144,35 @@ namespace mu2e {
 
       if ( diagLevel_ > 0 ) std::cout<<"[CaloDigiMaker::produce] begin" << std::endl;
 
-      ConditionsHandle<AcceleratorParams> accPar("ignored");
-      mbtime_ = accPar->deBuncherPeriod;
+      //get Event window and bunch timing info
+      art::Handle<EventWindowMarker> ewMarkerHandle;
+      event.getByLabel(ewMarkerTag_, ewMarkerHandle);
+      const EventWindowMarker& ewMarker(*ewMarkerHandle);
       
       auto caloShowerStepHandle = event.getValidHandle(caloShowerToken_);
       const auto& CaloShowerROs = *caloShowerStepHandle;
       
       auto caloDigiColl = std::make_unique<CaloDigiCollection>();
-      makeDigitization(CaloShowerROs, *caloDigiColl);
+      makeDigitization(CaloShowerROs, *caloDigiColl,ewMarker);
       event.put(std::move(caloDigiColl));
 
       if ( diagLevel_ > 0 ) std::cout<<"[CaloDigiMaker::produce] end" << std::endl;    
   }
 
   
   //-----------------------------------------------------------------------------------------------------------------------------
-  void CaloDigiMaker::makeDigitization(const CaloShowerROCollection& CaloShowerROs,CaloDigiCollection& caloDigiColl)
+  void CaloDigiMaker::makeDigitization(const CaloShowerROCollection& CaloShowerROs, CaloDigiCollection& caloDigiColl, const EventWindowMarker& ewMarker )
   {
       mu2e::GeomHandle<mu2e::Calorimeter> ch;
       calorimeter_ = ch.get();
 
+      ConditionsHandle<AcceleratorParams> accPar("ignored");
       ConditionsHandle<CalorimeterCalibrations> calorimeterCalibrations("ignored");
       if (calorimeter_->nCrystal()<1 || calorimeter_->caloInfo().getInt("nSiPMPerCrystal")<1) return;
-      
+ 
+      double mbtime    = (ewMarker.spillType() == EventWindowMarker::SpillType::onspill) ?  accPar->deBuncherPeriod : ewMarker.eventLength();      
       int nWaveforms   = calorimeter_->nCrystal()*calorimeter_->caloInfo().getInt("nSiPMPerCrystal");
-      int waveformSize = (mbtime_ - blindTime_ + endTimeBuffer_) / digiSampling_; 
+      int waveformSize = (mbtime - blindTime_ + endTimeBuffer_) / digiSampling_; 
```

Comment:
Ok reading through it again I think it will handle the end of the event properly, but you have fixed the digitization length to be 1695-blindTime+endTimeBuffer. It might be more straightforward to have digitizationStart and digitizationEnd instead as we do in the tracker

Reply:
When Richie made that that change ( to  digitizationStart and digitizationEnd ) there was a consensus the new way is the right way to think about it.  Also people objected to the word "blind" as a distraction because it suggested signal blinding.